TestStack.Dossier.Tests/BuildTests.cs

```diff
@@ -48,6 +49,32 @@ public void GivenBuilder_WhenCallingSetExplicitly_ShouldOverrideValues()
         }
 
         [Fact]
+        public void GivenBuilder_WhenCallingSetWithLambda_ShouldInvokeEachTime()
+        {
+            int counter = 2014;
+            var builder = new CustomerBuilder()
+                .Set(x => x.FirstName, "Pi")
+                .Set(x => x.LastName, "Lanningham")
+                .Set(x => x.YearJoined, () => counter++);
```

Comment:
It would be too hard to do the partially constructed object, you could do one that takes the builder so you can use a `Get` call thought?